hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/common/AbstractStreamWriteFunction.java

```diff
@@ -193,30 +174,28 @@ public void setOperatorEventGateway(OperatorEventGateway operatorEventGateway) {
   //  Utilities
   // -------------------------------------------------------------------------
 
-  private void restoreWriteMetadata() throws Exception {
-    boolean eventSent = false;
-    HoodieTimeline pendingTimeline = this.metaClient.getActiveTimeline().filterPendingExcludingCompaction();
-    for (WriteMetadataEvent event : this.writeMetadataState.get()) {
-      // Must filter out the completed instants in case it is a partial failover,
-      // the write status should not be accumulated in such case.
-      if (pendingTimeline.containsInstant(event.getInstantTime())) {
-        // Reset taskID for event
-        event.setTaskID(taskID);
-        // The checkpoint succeed but the meta does not commit,
-        // re-commit the inflight instant
-        this.eventGateway.sendEventToCoordinator(event);
-        LOG.info("Send uncommitted write metadata event to coordinator, task[{}].", taskID);
-        eventSent = true;
+  private void sendBootstrapEvent() throws Exception {
+    int attemptId = getRuntimeContext().getAttemptNumber();
+    if (attemptId <= 0) {
+      HoodieTimeline pendingTimeline = this.metaClient.getActiveTimeline().filterPendingExcludingCompaction();
+      // if the task is initially started, resend the pending event.
+      for (WriteMetadataEvent event : this.writeMetadataState.get()) {
```

Comment:
Write event from `writeMetadataState` should also be sent to coordinator for cases `attemptId` > 0, i.e., task failover?

Reply:
Ignore this. It seems ok here as the write event for failover tasks should be kept on coordinator.